File: Projects/Project1/generate.py
import os
import random
import numpy as np

folder_path = "./datasets"

if not os.path.exists(folder_path):
    os.makedirs(folder_path)

sizes = np.logspace(3, 7, num=50, dtype=int)

for size in sizes:
    random_data = random.sample(range(1, size + 1), size)
    filename = os.path.join(folder_path, f"{size}.txt") 

    with open(filename, "w") as file:
        for num in random_data:
            file.write(str(num) + ",")

File: Projects/Project1/generate.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cstdlib>
#include <ctime>
#include <string>
#include <direct.h>

int main() {
    std::string folder_path = "./datasets";

    // if (mkdir("./datasets") == -1) {
    //     std::cerr << "Error creating directory" << std::endl;
    //     return 1;
    // }

    srand(time(nullptr));

    for (int size = 5; size <= 10000000; size += 5) {
        std::vector<int> random_data;
        random_data.reserve(size);

        for (int i = 1; i <= size; ++i) {
            random_data.push_back(i);
        }

        // Shuffle the data randomly
        for (int i = size - 1; i > 0; --i) {
            int j = rand() % (i + 1);
            std::swap(random_data[i], random_data[j]);
        }

        std::string filename = folder_path + "/" + std::to_string(size) + ".txt";
        std::ofstream file(filename);

        if (!file) {
            std::cerr << "Error creating file: " << filename << std::endl;
            return 1;
        }

        for (int num : random_data) {
            file << num << ",";
        }

        file.close();
    }

    return 0;
}



